### PR TITLE
Adds a payment status

### PR DESCRIPTION
**Pull request by JordanValdma**

### CODE REVIEW

File: src/main/java/ee/tuleva/onboarding/payment/PaymentProviderCallbackService.java [hunk 1]
```diff
@@ -43,6 +43,7 @@ public void processToken(String serializedToken) {
               .amount(amount)
               .internalReference(internalReference.getUuid())
               .user(user)
+              .status(PaymentStatus.PENDING)
```
Comment (ErkoRisthein) [inline src/main/java/ee/tuleva/onboarding/payment/PaymentProviderCallbackService.java:46]: There might be a different status in the JWT token coming from Montonio

Comment (ErkoRisthein) [reply src/main/java/ee/tuleva/onboarding/payment/PaymentProviderCallbackService.java:46]: We should probably double check that the Montonio payment status is `finalized`

Comment (JordanValdma) [reply src/main/java/ee/tuleva/onboarding/payment/PaymentProviderCallbackService.java:46]: Done, thanks

File: src/main/java/ee/tuleva/onboarding/payment/PaymentProviderCallbackService.java [hunk 2]
```diff
@@ -35,20 +35,29 @@ public void processToken(String serializedToken) {
...
               .build();
 
       paymentRepository.save(paymentToBeSaved);
     }
   }
 
+  private boolean paymentWithReferenceExists(PaymentReference internalReference) {
+    return paymentRepository.findByInternalReference(internalReference.getUuid()).isEmpty();
+  }
```
Comment (ErkoRisthein) [inline src/main/java/ee/tuleva/onboarding/payment/PaymentProviderCallbackService.java:55]: `paymentWithReferenceExists == findByInternalReference isEmpty()`? Isn't it contradictory?

If it's empty then it _doesn't_ exist.

Comment (JordanValdma) [reply src/main/java/ee/tuleva/onboarding/payment/PaymentProviderCallbackService.java:55]: fixed, thanks

**Review by ErkoRisthein — APPROVED** [review]

**Pull request merged**